service/ui.go

```diff
@@ -312,3 +322,27 @@ func (uis *UIServer) LoggedError(w http.ResponseWriter, r *http.Request, code in
 		http.Error(w, err.Error(), code)
 	}
 }
+
+func (uis *UIServer) GetCommonViewData(w http.ResponseWriter, r *http.Request, needsUser, needsProject bool) ViewData {
+	viewData := ViewData{}
+	userCtx := GetUser(r)
+	if needsUser && userCtx == nil {
+		grip.Error("no user attached to request")
```

Comment:
Should these error cases return an error? This function could return `ViewData, err`.

Reply:
Deleted the early returns